src/warehouses/components/WarehouseInfo/WarehouseInfo.test.tsx

```diff
@@ -0,0 +1,114 @@
+import { WarehouseErrorCode } from "@dashboard/graphql";
+import { warehouse } from "@dashboard/warehouses/fixtures";
+import Wrapper from "@test/wrapper";
+import { fireEvent, render, screen } from "@testing-library/react";
+
+import WarehouseInfo from "./WarehouseInfo";
+
+const mockOnChange = jest.fn();
+
+const invalidNameError = [
+  {
+    __typename: "WarehouseError" as const,
+    code: WarehouseErrorCode.INVALID,
+    field: "name",
+    message: "Name is required",
+  },
+];
+
+const invalidEmailError = [
+  {
+    __typename: "WarehouseError" as const,
+    code: WarehouseErrorCode.INVALID,
+    field: "email",
+    message: "Invalid email format",
+  },
+];
+
+describe("WarehouseInfo", () => {
+  beforeEach(() => {
+    mockOnChange.mockClear();
+  });
+
+  it("renders warehouse name and email fields", () => {
+    render(
+      <Wrapper>
+        <WarehouseInfo data={warehouse} disabled={false} errors={[]} onChange={mockOnChange} />
+      </Wrapper>,
+    );
+
+    const nameInput = screen.getByTestId("warehouse-name-input").querySelector("input");
+    const emailInput = screen.getByTestId("company-email-input").querySelector("input");
+
+    expect(nameInput).toBeInTheDocument();
+    expect(emailInput).toBeInTheDocument();
+    expect(nameInput).toHaveValue(warehouse.name);
+    expect(emailInput).toHaveValue(warehouse.email);
+  });
+
+  it("calls onChange when warehouse name is modified", () => {
+    render(
+      <Wrapper>
+        <WarehouseInfo data={warehouse} disabled={false} errors={[]} onChange={mockOnChange} />
+      </Wrapper>,
+    );
+
+    const nameInput = screen.getByTestId("warehouse-name-input").querySelector("input");
+
+    expect(nameInput).toBeInTheDocument();
+    fireEvent.change(nameInput, { target: { value: "Updated Warehouse" } });
```

Comment:
I think its fine to 

```
    fireEvent.change(nameInput!, { target: { value: "Updated Warehouse" } });
```

Because this is test and because L56 will fail if input is not found